File: Linked_list_Reverse.py
Iterative solution:
// Time Complexity : O(n)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : None


// Your code here along with comments explaining your approach
we do this problem using three pointers.we start by adjusting these three pointers to get the linked list reversal.

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        prev=None
        curr=head
        while curr!=None:
            next1=curr.next
            curr.next=prev
            prev=curr
            curr=next1
        return prev


Recursive solution:
// Time Complexity : O(n)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : None


// Your code here along with comments explaining your approach
Inorder to reverse the linked list we traverse to the end of the linked list and point the next of next element of the current element to the current element.In this manner we do this for all the nodes.


class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head==None or head.next==None:
            return head
        rev=self.reverseList(head.next)
        head.next.next=head
        head.next=None
        return rev

File: Linked_List_Reverse.cs
using System;
namespace Algorithms
{
    /// Time Complexity : O(n) 
    // Space Complexity :O(1)
    // Did this code successfully run on Leetcode :Yes
    // Any problem you faced while coding this :  Referred Online
    public class Linked_List_Reverse
    {
        public class ListNode
        {
            public int val;
            public ListNode next;
            public ListNode(int val = 0, ListNode next = null)
            {
                this.val = val;
                this.next = next;
            }
        }

        public ListNode ReverseList(ListNode head)
        {
            if (head == null) return head;
            ListNode prev = null;
            ListNode current = head;
            ListNode fast = head.next;
            while (fast != null)
            {
                current.next = prev;
                prev = current;
                current = fast;
                fast = fast.next;
            }
            current.next = prev;
            return current;

        }

        public ListNode ReverseList_Recursive(ListNode head)
        {
            if (head == null || head.next == null) return head;

            ListNode reversed = ReverseList_Recursive(head.next);
            head.next.next = head;
            head.next = null;
            return reversed;
        }
    }
}
